src/pages-and-resources/discussions/data/thunks.js

```diff
@@ -0,0 +1,25 @@
+import { getDiscussionApps } from './api';
+import { addModels } from '../../../generic/model-store';
+import {
+  FAILED, fetchAppsSuccess, LOADING, updateStatus,
+} from './slice';
+
+/* eslint-disable import/prefer-default-export */
+export function fetchApps(courseId) {
```

Comment:
This function's job is to handle the asynchronous process of getting app data.  Along the way, it updates the redux store's state so that the app can show the user appropriate feedback.